### PR TITLE
feat: add navbar component

### PR DESCRIPTION
**Pull request by satyam73**
Date: 25-02-2024

Developer Name: Satyam Bajpai

---

## Issue Ticket Number
- #42 
## Description

Created a navbar component.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots

https://github.com/Real-Dev-Squad/skill-tree-frontend/assets/85296770/ddea3bb3-fa4d-41e3-8f47-fe7275875cde

image
> Desktop view

image
> Tab view

image
> Mobile view

## Test Coverage
image

### CODE REVIEW

File: public/github-logo.png
Comment (sahsisunny) [inline public/github-logo.png:1]: we can use react icons for this.

File: src/components/common/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                <button onClick={toggleNavbarLinksVisibility} className="hamburger lg:hidden">
+                    <GiHamburgerMenu className="text-white font-bold" size={30} />
+                </button>
+                <nav className="navbar__desktop max-lg:hidden visible">
+                    <ul className="flex gap-10 items-center max-lg:flex-col">{navbarItemsMapping}</ul>
+                </nav>
+                <a data-testid="signin-button" className="ml-auto decoration-none" href={"login url will come here"}>
+                    <span className="flex gap-2 items-center text-base max-sm:text-sm max-sm:font-bold font-semibold w-fit border text-white rounded-md py-2 p-2 ">
+                        <span>Sign in with Github</span>
+                        <IoLogoGithub size={25} />
```
Comment (sahsisunny) [inline src/components/common/Navbar/Navbar.tsx:38]: If you are using the react icon here then why do you add svg for the GitHub logo?

File: tailwind.config.js [hunk 2]
```diff
@@ -23,6 +23,9 @@ module.exports = {
             },
         },
         colors: {
+            primary: "#041187",
```
Comment (satyam73) [inline tailwind.config.js:26]: primary: `#041187`
secondary: `#E30062`
contrast: `#85DA6B`

File: package.json [hunk 3]
```diff
@@ -7,8 +7,8 @@
         "build:production": "cp ./env/.env.production .env && next build",
         "build:staging": "cp ./env/.env.staging .env && next build",
         "start": "next start",
-        "dev:server": "next dev -p 3000",
-        "dev:reverse-ssl": "local-ssl-proxy --source 443 --target 3000",
+        "dev:server": "next dev -p 5000",
+        "dev:reverse-ssl": "local-ssl-proxy --source 443 --target 5000",
```
Comment (sahsisunny) [inline package.json:11]: why?

**Comment by ajoykumardas12** [comment]
In the video the navbar disappears after sign in, didn't get that.
Also, still showing 'sign in with github` after sign in.

File: src/components/common/Navbar/navbar.constants.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { RDS_BACKEND_URL } from "@/constants/urls";
+import { NavbarLink } from "./navbar.types";
+
+// ?v2=true so that we get rds-session-v2 cookie from RDS backend
+export const SIGNIN_URL = `${RDS_BACKEND_URL}/auth/github/login?redirectURL=${process.env.NEXT_PUBLIC_SKILL_TREE_SITE}?v2=true`;
```
Comment (ajoykumardas12) [inline src/components/common/Navbar/navbar.constants.ts:5]: `?` used twice in query.

File: src/components/common/Navbar/navbar.constants.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+export const NAVBAR_LINKS: NavbarLink[] = [
+    {
+        id: "welcome-site",
```
Comment (ajoykumardas12) [inline src/components/common/Navbar/navbar.constants.ts:9]: `...-site-link` is better as id of the links I think.

File: src/components/common/Navbar/navbar.constants.ts (hunk 4 above)
Comment (satyam73) [reply src/components/common/Navbar/navbar.constants.ts:5]: Yes it's intended behaviour

**Comment by satyam73** [comment]
> In the video the navbar disappears after sign in, didn't get that.

It redirected to staging-skilltree.realdevsquad.com since I haven't changed in my dev environment env file while making video, this isn't a problem and will work accordingly in different environments.

As of now I haven't added name and profile picture in navbar, this will be handled separately in next PR.

**Review by sahsisunny — APPROVED** [review]

**Review by ajoykumardas12 — APPROVED** [review]

**Review by VinayakaHegade — APPROVED** [review]
Looks good to me.

**Review by ajoykumardas12 — DISMISSED** [review]

**Review by VinayakaHegade — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Review by VinayakaHegade — APPROVED** [review]

**Review by ajoykumardas12 — APPROVED** [review]

**Pull request merged**